sessionctx/variable/statusvar.go

```diff
@@ -42,8 +42,11 @@ type Statistics interface {
 	Stats(*SessionVars) (map[string]interface{}, error)
 }
 
-// RegisterStatistics registers statistics.
-func RegisterStatistics(s Statistics) {
+// RegisterStatistics is exposed to avoid holding a reference in global variables, when using TiDB as a library.
+var RegisterStatistics = registerStatistics
+
+// registerStatistics registers statistics.
+func registerStatistics(s Statistics) {
```

Comment:
I think we should add a function like `UnregisterStatistics` and called it when domain or ddl is closed. It seems memory leak also happened in TiDB server, because `BootstrapSession` open `domain` twice.

Reply:
there's no id for `Statistics`, we can't find the one we appended (maybe by memory address?)

Reply:
Directly comparing interface?